airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -41,6 +42,18 @@ def _try_to_boolean(value: Any):
     return value
 
 
+def _maybe_add_prefix(val):
+    """
+    From Airflow 2.3 onward, there is no longer a need to add the `extra__<conn type>__`
```

Comment:
Next provider release cycle is going to set min Airflow version 2.3 for all providers:
https://github.com/apache/airflow#release-process-for-providers

Since the next cycle is on October any way maybe we should hold this PR till we adjust the min Airflow version?

Reply:
sure, that's fine

Reply:
when do you think we will bump the min version?  will there be time to add this after bumping and before release?

Reply:
The bump can happen only after 11-Oct because we might need to make ad-hoc release to fix something urgently. Right after 11-Oct we can bump all providers and then proceed.

The reason I suggested it is because we usually do 1 release per month so the next release is expected around the end of October thus the current "backport" code for 2.2 will never reach to the user anyway.

We can also merge this first and do cleanup after.. That is OK.